core/src/main/scala/org/scalarelational/Session.scala

```diff
@@ -12,12 +12,13 @@ case class Session(datastore: Datastore, var inTransaction: Boolean = false) {
   private var _disposed = false
   private var _connection: Option[Connection] = None
 
-  def hasConnection = _connection.nonEmpty
-  def connection = _connection match {
+  def hasConnection: Boolean = _connection.nonEmpty
+
+  def connection: Connection = _connection match {
     case _ if disposed => throw new RuntimeException("Session is disposed.")
     case Some(c) => c
     case None => {
-      val c = datastore.dataSource.getConnection
+      val c = datastore.dataSource.get.getConnection
```

Comment:
wouldn't this be better as: `val c = datastore.dataSource.map(_.getConnection)` so you don't then have to wrap it in an Option and then just call `c.get` at the end?